src/hooks/useInfnityScroll.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useRef } from "react";
+
+export interface FetchMoreProps {
+  hasMore: boolean;
+  onFetchMore: () => void;
+  loading?: boolean;
+}
+
+export const useInfnityScroll = (fetchMore: FetchMoreProps | undefined) => {
+  const observerTarget = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (!fetchMore || !observerTarget) {
+      return;
+    }
+
+    const observer = new IntersectionObserver(
```

Comment:
**issue**: that means you create new instance on each render, without cleaning it - i expect memory leak here. If you create a new instance, first you need to clean previous one